instrumentation/github.com/syndtr/goleveldb/leveldb/splunkleveldb/iterator.go

```diff
@@ -32,10 +32,10 @@ type iter struct {
 // called.
 func WrapIterator(it iterator.Iterator, opts ...Option) iterator.Iterator {
 	c := newConfig(opts...)
-	sso := c.mergedSpanStartOptions(
+	sso := c.MergedSpanStartOptions(
 		trace.WithAttributes(semconv.DBOperationKey.String("Iterator")),
 	)
-	_, span := c.resolveTracer().Start(c.ctx, "Iterator", sso...)
+	_, span := c.ResolveTracer(c.ctx).Start(c.ctx, "Iterator", sso...)
```

Comment:
I find it awkward to have the the `ctx` in the config struct. 
I propose to have two methods in the common config:
- ResolveTracer
- ResolveTracerContext (when some call have context)

Same for `WithSpan`. 

Then if the instrumented library used context we use `ResolveTracerContext` and `WithSpanContext` (e.g. for `http`) if not then we simply use `ResolveTracerContext` and `WithSpan` (e.g. for `leveldb`).

Calling `trace.SpanFromContext(ctx)` is useless if we know that it will never return anything.

If you disagree I would rather put `context.Background()` in all places instead of `cfg.ctx`. Current code can give a false impression that can be some real context.

Reply:
> I find it awkward to have the the `ctx` in the config struct.

Sure, agreed. However, this is an artifact of the underlying library not accepting a context as an argument. The instrumentation then needs to accept this an apply it as needed.

> I propose to have two methods in the common config:
> 
> * ResolveTracer
> * ResolveTracerContext (when some call have context)

`ResolveTracer` is equivalent to just `Config.Tracer`. If you don't have a context to extract a `TracerProvider` from it should not be used.

> Same for `WithSpan`.

The context passed to `WithSpan` is updated to include the span and passed to the closure. Having the ability to set the parent is going to be important for more than just span propagation. Like the rest of the Go ecosystem, having a context as a first argument is a good idea in that it keeps context propagation explicit and adding `*Context` type functions are useful only in that they allow backwards compatibility for context propagation. Having one from the start seems to be the opposite of idiomatic Go.

> If you disagree I would rather put `context.Background()` in all places instead of `cfg.ctx`. Current code can give a false impression that can be some real context.

That can in fact be a real context, with a span embedded in it. Passing that context as included in this PR ensures the correct user specified `TracerProvider` is used an heredity is preserved.